### ISSUE TITLE
Bump to DuckDB 0.10.3

### ISSUE THREAD

**Issue by allistercsmith** [duckdb/duckdb-rs#322]
I'm hoping to use the new [Deltalake extension](https://github.com/duckdb/duckdb_delta), which looks to require DuckDB v0.10.3, whereas this wrapper is currently pinned to 0.10.2.

Attempting to use the `delta_scan` function right now results in the following error:

```
Error: Catalog Error: Table Function with name delta_scan does not exist!
Did you mean "index_scan"?
```

**Comment by allistercsmith** [comment]
@Mause I see you recently [bumped versions](https://github.com/duckdb/duckdb-rs/pull/298). I attempted to follow a similar approach to your PR by bumping the version references to 0.10.3 (mainly in `libduckdb-sys/upgrade.sh`), but I'm seeing a compilation error during the sanity checks stage:

```
Undefined symbols for architecture arm64:
  "vtable for duckdb::HuggingFaceFileSystem", referenced from:
      duckdb::HuggingFaceFileSystem::HuggingFaceFileSystem() in libduckdb.a[84](653b37f8bf82ced4-httpfs_extension.o)
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

A bit lost on how to proceed from here.

**Comment by Maxxen** [comment]
We just released DuckDB v1.0.0 so we're probably going to skip v0.10.3. There's a couple of open PR's I want to review and merge first but expect a bump later this week.

**Comment by Mytherin** [comment]
DuckDB v1.0.0 is now published
